pkg/charts/utils.go

```diff
@@ -160,12 +160,17 @@ func generateChartValues(config *ciliumv1alpha1.NetworkConfig, network *extensio
 		globalConfig.LocalRedirectPolicy.Enabled = true
 	}
 
+	// disable PSPs if it's disabled in the shoot
+	if helper.IsPSPDisabled(cluster.Shoot) {
+		globalConfig.Psp.Enabled = false
+	}
+
 	if config == nil {
 		return requirementsConfig, globalConfig, nil
 	}
 
 	// check if PSPs are enabled
-	if config.PSPEnabled != nil {
+	if config.PSPEnabled != nil && globalConfig.Psp.Enabled {
```

Comment:
Please adapt the comment and reorder the checks of the `if` condition.
Reasoning: It is important to note that we do not want to enable the pod security policy if it was already disabled in the step before. Hence, the switch goes just in one direction. At least for me it is easier to comprehend if the check is written the other way around and the comment mentions this.